syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -408,36 +415,45 @@ func newPackageFromMavenData(pomProperties pkg.PomProperties, pomProject *pkg.Po
 }
 
 func packageIdentitiesMatch(p pkg.Package, parentPkg *pkg.Package) bool {
-	// the name/version pair matches...
-	if uniquePkgKey(&p) == uniquePkgKey(parentPkg) {
-		return true
-	}
+	childMetadata, childOk := p.Metadata.(pkg.JavaMetadata)
+	parentMetadata, parentOk := parentPkg.Metadata.(pkg.JavaMetadata)
+	if !childOk || !parentOk {
+		switch {
+		case !childOk:
+			log.WithFields("package", p.String()).Debug("unable to extract java metadata to check for matching package identity for package: %s", p.Name)
```

Comment:
both of these should probably be trace lines

Reply:
resolved with [29a3a68](https://github.com/anchore/syft/pull/2033/commits/29a3a68644609f7fec877070bae7c4db30aee039)